contracts/superclasses/Posts.sol

```diff
@@ -323,33 +332,45 @@ contract Posts is KetlGuarded {
     // Fetch post or comment
     Post memory post = commentId == 0
       ? posts[feedId][postId]
-      : comments[feedId][postId][commentId];
+      : comments[feedId][postId][commentId - 1]; // -1 because commentIds starts with 1
     // Check if post or comment exists
-    require(post.author != address(0), "Post or comment not found");
+    require(post.sender != address(0), "Post or comment not found");
     // Get old reaction if it exists
     Reaction memory oldReaction = usersToReactions[feedId][postId][commentId][
       sender
     ];
     // Check if reaction already exists
-    require(oldReaction.sender == address(0), "Reaction already exists");
+    require(
+      oldReaction.reactionType != reactionType,
+      "Reaction of this type already exists"
+    );
+    if (oldReaction.reactionType != 0) {
+      this.removeReaction(
```

Comment:
There should be a way to not use `this`. Check the visibility of the `removeReaction` function and if there are any duplicate function names

Reply:
Change `external` to `public` at `removeReaction` and remove `this`